lib/postgrex/protocol.ex

```diff
@@ -2305,7 +2317,9 @@ defmodule Postgrex.Protocol do
         recv_execute(s, status, query, rows, buffer)
 
       {:disconnect, _, _} = dis ->
-        dis
+        with {_, %{reason: :closed} = err, s} <- dis do
+          {:disconnect, %{err | reason: :execute_closed}, s}
```

Comment:
The reason I do an exclusive condition i.e. ignore `:execute_closed` instead of an inclusive one i.e. accept `:bind_closed` is because the failure most of the time will come on the `mod.send(sock, data)` and not as part of the `recv_bind` part

Reply:
This looks good to me but perhaps I would write two clauses:

```
{:disconnect, %{reason: :closed}, s} ->

{:disconnect, _, _} = dis -> dis
```

Reply:
Actually, now that ì think about it, this change means we are now doing too broad checks, so we have to replace closed by execute_closed, right? If so, I'd prefer if we didn't have to do this and instead we move `handle_disconnect_retry` lower in the stack. I will drop another comment.